app/helpers/admin/url_helper.rb

```diff
@@ -0,0 +1,44 @@
+module Admin::UrlHelper
+  require 'uri'
+
+  def build_url(base_url, page)
+    if page.class.name == 'Page'
+      "#{base_url}#{page.path}"
+    else
+      path = lookup_page_path(page)
+      path ? "#{base_url}/#{path}/#{page.slug}" : nil
+    end
+  end
+
+  def extract_base_url(url)
+    uri = URI.parse(url)
+    host = uri.host
+    scheme = uri.scheme
+    return "#{scheme}://#{host}:#{uri.port}" if host&.include?('localhost')
+
+    "#{scheme}://#{host}"
+  end
+
+  def generate_page_url(url, page)
+    base_url = extract_base_url(url)
+    build_url(base_url, page)
+  end
+
+  def lookup_page_path(page)
+    PAGE_TYPES[page.class.name.to_sym]
+  end
+
+  PAGE_TYPES = {
```

Comment:
My only change would be to pull this out and place it in the cd.org app and then again in the fest app. We should be keeping "Types" separate since they belong to the applications that use trustycms - not trusty itself.